ipc/nodepart/arwenDriver.go

```diff
@@ -0,0 +1,308 @@
+package nodepart
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/common"
+	"github.com/ElrondNetwork/arwen-wasm-vm/ipc/logger"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+var _ vmcommon.VMExecutionHandler = (*ArwenDriver)(nil)
+
+// ArwenDriver manages the execution of the Arwen process
+type ArwenDriver struct {
+	nodeLogger     logger.Logger
+	blockchainHook vmcommon.BlockchainHook
+	vmType         []byte
+	blockGasLimit  uint64
+	gasSchedule    map[string]map[string]uint64
+
+	arwenInputRead   *os.File
+	arwenInputWrite  *os.File
+	arwenOutputRead  *os.File
+	arwenOutputWrite *os.File
+	arwenLogRead     *os.File
+	arwenLogWrite    *os.File
+	command          *exec.Cmd
+	part             *NodePart
+}
+
+// NewArwenDriver creates a new driver
+func NewArwenDriver(
+	nodeLogger logger.Logger,
+	blockchainHook vmcommon.BlockchainHook,
+	vmType []byte,
+	blockGasLimit uint64,
+	gasSchedule map[string]map[string]uint64,
+) (*ArwenDriver, error) {
+	driver := &ArwenDriver{
+		nodeLogger:     nodeLogger,
+		blockchainHook: blockchainHook,
+		vmType:         vmType,
+		blockGasLimit:  blockGasLimit,
+		gasSchedule:    gasSchedule,
+	}
+
+	err := driver.startArwen()
+	return driver, err
+}
+
+func (driver *ArwenDriver) startArwen() error {
+	driver.nodeLogger.Info("ArwenDriver.startArwen()")
+
+	driver.resetPipeStreams()
+
+	arwenPath, err := driver.getArwenPath()
+	if err != nil {
+		return err
+	}
+
+	arguments, err := common.PrepareArguments(driver.vmType, driver.blockGasLimit, driver.gasSchedule, logger.LogDebug)
+	if err != nil {
+		return err
+	}
+
+	driver.command = exec.Command(arwenPath, arguments...)
+	driver.command.ExtraFiles = []*os.File{driver.arwenInputRead, driver.arwenOutputWrite, driver.arwenLogWrite}
+
+	arwenStdout, err := driver.command.StdoutPipe()
+	if err != nil {
+		return err
+	}
+
+	arwenStderr, err := driver.command.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	err = driver.command.Start()
+	if err != nil {
+		return err
+	}
+
+	driver.part, err = NewNodePart(driver.nodeLogger, driver.arwenOutputRead, driver.arwenInputWrite, driver.blockchainHook)
+	if err != nil {
+		return err
+	}
+
+	driver.continuouslyCopyArwenLogs(arwenStdout, arwenStderr)
```

Comment:
`continuouslyCopyArwenLogs` also captures Arwen's `stdout` and `stderr` in addition to `arwenLog` (the read-end of the logging pipe). 

`Stdout` should contain nothing in production (though, it does, some prints from Wasmer, `Init global opcodes` or something like that). `Stderr` should contain nothing in the happy case - but it does in case of panics. So we have to read it too.